### DISCUSSION TITLE
UseCase 설계

### DISCUSSION THREAD

**Discussion by softpeanut** [Team-ComIT/SimTong-Backend#42] [의논하기]
좋은 소프트웨어를 설계하기 위해서는 응집도를 높이고 결합도를 낮추어야 한다.
- **응집도**
> 하나의 모듈(클래스) 내부에 존재하는 구성 요소들의 기능적 관련성으로 정의할 수 있다.
> 응집도가 높은 모듈은 하나의 책임에 집중해 독립성이 높아지게 된다.
- **결합도**
> 모듈(클래스)간의 상호 의존 정도를 나타내는 지표로 정의할 수 있다.
> 결합도가 낮으면 모듈간의 상호 의존성이 줄어든다.

**Comment by khcho0125** [comment]
음 그렇다면 하나의 UseCase로 관리하는게 좋겠네요 :)

**Reply by softpeanut** [reply]
그렇게 생각하신 이유가 있으신가요?

**Comment by khcho0125** [comment]
UseCase를 하나의 UseCase로 관리한다면 UseCase Test의 의존성이나 변수들이 매우 많아져 TestCase작성이 어려울 것으로 판단됩니다.
여러개의 UseCase를 사용하는게 더욱 효과적일 것이라고 생각합니다.
